pubspec.yaml

```diff
@@ -9,7 +9,7 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  flutter_svg: ^0.17.4
+  flutter_svg: ^0.18.0
```

Comment:
flutter_svg is not using stable flutter branch. I don't think we need to upgrade it from 0.17.4 to 0.18.0.

Reply:
According to https://github.com/dnfield/flutter_svg/pull/359 it should work because the new mixin is in stabel ages ago.

Reply:
When I using your branch to build, it still got the error which the Flutter SDK requires >= 1.18.0-6.0.pre < 2.0.0 . 
flutter_svg is not using the stable flutter channel in the 0.18.0 versioni. 
```
 % flutter pub get
The current Flutter SDK version is 1.17.5.

Because flag depends on flutter_svg >=0.18.0 which requires Flutter SDK version >=1.18.0-6.0.pre <2.0.0, version solving failed.
---- Log transcript ----
FINE: Pub 2.8.4
MSG : Resolving dependencies...
SLVR: fact: flag is 3.0.1
SLVR: derived: flag
SLVR: fact: flag depends on flutter any from sdk
SLVR: fact: flag depends on flutter_svg ^0.18.0
SLVR: fact: flag depends on flutter_test any from sdk
SLVR:   selecting flag
SLVR:   derived: flutter_test any from sdk
SLVR:   derived: flutter_svg ^0.18.0
SLVR:   derived: flutter any from sdk
IO  : Get versions from https://pub.dartlang.org/api/packages/flutter_svg.
IO  : HTTP GET https://pub.dartlang.org/api/packages/flutter_svg
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: macos
    | X-Pub-Command: get
    | X-Pub-Session-ID: 8DF196C0-B177-465A-85DF-AC6CBEED1089
    | X-Pub-Environment: flutter_bot:flutter_cli:get
    | X-Pub-Reason: direct
    | user-agent: Dart pub 2.8.4
IO  : HTTP response 200 OK for GET https://pub.dartlang.org/api/packages/flutter_svg
    | took 0:00:00.670068
    | transfer-encoding: chunked
    | date: Sat, 01 Aug 2020 11:21:57 GMT
    | content-encoding: gzip
    | vary: Accept-Encoding
    | strict-transport-security: max-age=31536000; preload
    | via: 1.1 google
    | content-type: application/json; charset="utf-8"
    | x-frame-options: SAMEORIGIN
    | x-xss-protection: 1; mode=block
    | x-content-type-options: nosniff
    | server: dart:io with Shelf
IO  : Get versions from https://pub.dartlang.org/api/packages/path_drawing.
IO  : Get versions from https://pub.dartlang.org/api/packages/xml.
IO  : Get versions from https://pub.dartlang.org/api/packages/vector_math.
IO  : Get versions from https://pub.dartlang.org/api/packages/meta.
IO  : HTTP GET https://pub.dartlang.org/api/packages/path_drawing
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: macos
    | X-Pub-Command: get
    | X-Pub-Session-ID: 8DF196C0-B177-465A-85DF-AC6CBEED1089
    | X-Pub-Environment: flutter_bot:flutter_cli:get
    | X-Pub-Reason: direct
    | user-agent: Dart pub 2.8.4
IO  : HTTP GET https://pub.dartlang.org/api/packages/xml
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: macos
    | X-Pub-Command: get
    | X-Pub-Session-ID: 8DF196C0-B177-465A-85DF-AC6CBEED1089
    | X-Pub-Environment: flutter_bot:flutter_cli:get
    | X-Pub-Reason: direct
    | user-agent: Dart pub 2.8.4
IO  : HTTP GET https://pub.dartlang.org/api/packages/vector_math
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: macos
    | X-Pub-Command: get
    | X-Pub-Session-ID: 8DF196C0-B177-465A-85DF-AC6CBEED1089
    | X-Pub-Environment: flutter_bot:flutter_cli:get
    | X-Pub-Reason: direct
    | user-agent: Dart pub 2.8.4
IO  : HTTP GET https://pub.dartlang.org/api/packages/meta
    | Accept: application/vnd.pub.v2+json
    | X-Pub-OS: macos
    | X-Pub-Command: get
    | X-Pub-Session-ID: 8DF196C0-B177-465A-85DF-AC6CBEED1089
    | X-Pub-Environment: flutter_bot:flutter_cli:get
    | X-Pub-Reason: direct
    | user-agent: Dart pub 2.8.4
SLVR:   fact: flutter_test 0.0.0 from sdk depends on flutter any from sdk
SLVR:   fact: flutter_test 0.0.0 from sdk depends on test_api 0.2.15
SLVR:   fact: flutter_test 0.0.0 from sdk depends on path 1.6.4
SLVR:   fact: flutter_test 0.0.0 from sdk depends on image 2.1.12
SLVR:   fact: flutter_test 0.0.0 from sdk depends on quiver 2.1.3
SLVR:   fact: flutter_test 0.0.0 from sdk depends on stack_trace 1.9.3
SLVR:   fact: flutter_test 0.0.0 from sdk depends on vector_math 2.0.8
SLVR:   fact: flutter_test 0.0.0 from sdk depends on archive 2.0.13
SLVR:   fact: flutter_test 0.0.0 from sdk depends on args 1.6.0
SLVR:   fact: flutter_test 0.0.0 from sdk depends on async 2.4.1
SLVR:   fact: flutter_test 0.0.0 from sdk depends on boolean_selector 2.0.0
SLVR:   fact: flutter_test 0.0.0 from sdk depends on charcode 1.1.3
SLVR:   fact: flutter_test 0.0.0 from sdk depends on collection 1.14.12
SLVR:   fact: flutter_test 0.0.0 from sdk depends on convert 2.1.1
SLVR:   fact: flutter_test 0.0.0 from sdk depends on crypto 2.1.4
SLVR:   fact: flutter_test 0.0.0 from sdk depends on matcher 0.12.6
SLVR:   fact: flutter_test 0.0.0 from sdk depends on meta 1.1.8
SLVR:   fact: flutter_test 0.0.0 from sdk depends on petitparser 2.4.0
SLVR:   fact: flutter_test 0.0.0 from sdk depends on source_span 1.7.0
SLVR:   fact: flutter_test 0.0.0 from sdk depends on stream_channel 2.0.0
SLVR:   fact: flutter_test 0.0.0 from sdk depends on string_scanner 1.0.5
SLVR:   fact: flutter_test 0.0.0 from sdk depends on term_glyph 1.1.0
SLVR:   fact: flutter_test 0.0.0 from sdk depends on typed_data 1.1.6
SLVR:   fact: flutter_test 0.0.0 from sdk depends on xml 3.6.1
SLVR:     selecting flutter_test 0.0.0 from sdk
SLVR:     derived: xml 3.6.1
SLVR:     derived: typed_data 1.1.6
SLVR:     derived: term_glyph 1.1.0
SLVR:     derived: string_scanner 1.0.5
SLVR:     derived: stream_channel 2.0.0
SLVR:     derived: source_span 1.7.0
SLVR:     derived: petitparser 2.4.0
SLVR:     derived: meta 1.1.8
SLVR:     derived: matcher 0.12.6
SLVR:     derived: crypto 2.1.4
SLVR:     derived: convert 2.1.1
SLVR:     derived: collection 1.14.12
SLVR:     derived: charcode 1.1.3
SLVR:     derived: boolean_selector 2.0.0
SLVR:     derived: async 2.4.1
SLVR:     derived: args 1.6.0
SLVR:     derived: archive 2.0.13
SLVR:     derived: vector_math 2.0.8
SLVR:     derived: stack_trace 1.9.3
SLVR:     derived: quiver 2.1.3
SLVR:     derived: image 2.1.12
SLVR:     derived: path 1.6.4
SLVR:     derived: test_api 0.2.15
SLVR:     fact: flutter_svg >=0.18.0 requires Flutter SDK version >=1.18.0-6.0.pre <2.0.0
SLVR:     conflict: flutter_svg >=0.18.0 requires Flutter SDK version >=1.18.0-6.0.pre <2.0.0
SLVR:     ! flutter_svg >=0.18.0 is satisfied by flutter_svg ^0.18.0
SLVR:     ! which is caused by "flag depends on flutter_svg ^0.18.0"
SLVR:     ! thus: version solving failed
SLVR: Version solving took 0:00:00.843327 seconds.
    | Tried 1 solutions.
FINE: Resolving dependencies finished (0.8s).
ERR : The current Flutter SDK version is 1.17.5.
    |
    | Because flag depends on flutter_svg >=0.18.0 which requires Flutter SDK version >=1.18.0-6.0.pre <2.0.0, version solving failed.
FINE: Exception type: SolveFailure
FINE: package:pub/src/solver/version_solver.dart 312:5   VersionSolver._resolveConflict
    | package:pub/src/solver/version_solver.dart 133:27  VersionSolver._propagate
    | package:pub/src/solver/version_solver.dart 97:11   VersionSolver.solve.<fn>
    | ===== asynchronous gap ===========================
    | dart:async                                         Future.catchError
    | package:pub/src/utils.dart 113:52                  captureErrors.wrappedCallback
    | package:stack_trace                                Chain.capture
    | package:pub/src/utils.dart 126:11                  captureErrors
    | package:pub/src/command_runner.dart 191:13         PubCommandRunner.runCommand
---- End log transcript ----
Running "flutter pub get" in flag_flutter...
pub get failed (1; ---- End log transcript ----)
```

Reply:
Would you change the flutter_svg version back to 0.17.4 so I can merge this PR into master branch?

Reply:
That's ok, I will create some other branch to fix this.